SimpleJsonConverter cannot convert nested jason in SimpleJsonConverter.java on kafka-connect-redis-kafka-3-1-4.0.0

# Issue Guidelines

### issue
this is my json

`{
  "group_name": "bogFan",
  "vendor_id": 32,
  "username": "351861861",
  "time": "12:27:12.4455667",
  "users": [
    {
      "my_id": 32,
      "your_id": 31,
      "requests": [
        {
          "sand_code": 321,
          "age_code": 311
        },
        {
          "sand_code": 322,
          "age_code": 312
        }
      ]
    }
  ],
  "create_date": "2023-01-29 12:27"
}`

when I want to convert it and send it to redis, I get this message:

`org.apache.kafka.connect.errors.ConnectException: org.apache.kafka.connect.errors.DataException: Mismatching schema.`

the error come from this file

https://github.com/lensesio/kafka-connect-common/blob/master/src/main/java/com/datamountaineer/streamreactor/connect/json/SimpleJsonConverter.java

line 172 and 217

SimpleJsonConverter cannot convert nested jason, for more details it cannot convert `users `in my json.

for more details in that file in line `213 ` 
` if (struct.schema() != schema)
            throw new DataException("Mismatching schema.");`

cause  throw exception while this code work for simple json but not my json.


### What version of the Stream Reactor are you reporting this issue for?
kafka-connect-redis-kafka-3-1-4.0.0
I use kafka version 3.3-IV3

### Are you running the correct version of Kafka/Confluent for the Stream reactor release?
yes

### Do you have a supported version of the data source/sink .i.e Cassandra 3.0.9?
no

### Have you read the [docs](https://docs.lenses.io/connectors/)?
yes

### What is the expected behaviour?
it should convert my json noirmaly in save it on redis

### What was observed?
connector task get fails

### What is your connector properties configuration (my-connector.properties)?
this is connector config
`{
	"connector.class": "com.datamountaineer.streamreactor.connect.redis.sink.RedisSinkConnector",
	"errors.log.include.messages": "true",
	"connect.redis.port": "6379",
	"tasks.max": "1",
	"topics": "users-data",
	"connect.redis.host": "redis-master",
	"key.converter.schemas.enable": "false",
	"connect.redis.kcql": "INSERT INTO Users- SELECT * from users-data PK vendor_id TTL=7200",
	"name": "redis-sink",
	"value.converter.schemas.enable": "false",
	"value.converter": "org.apache.kafka.connect.storage.StringConverter",
	"errors.log.enable": "true",
	"key.converter": "org.apache.kafka.connect.json.JsonConverter"
}`


### Please provide full log files (redact and sensitive information)
I see this error

```
at com.datamountaineer.streamreactor.connect.json.SimpleJsonConverter.convertToJson(SimpleJsonConverter.java:217)
at com.datamountaineer.streamreactor.connect.json.SimpleJsonConverter.convertToJson(SimpleJsonConverter.java:172)
Caused by: org.apache.kafka.connect.errors.ConnectException: org.apache.kafka.connect.errors.DataException: Mismatching schema.
```

--- a/kafka-connect-common/src/main/java/com/datamountaineer/streamreactor/connect/json/SimpleJsonConverter.java
+++ b/kafka-connect-common/src/main/java/com/datamountaineer/streamreactor/connect/json/SimpleJsonConverter.java
@@ -210,7 +210,7 @@ public class SimpleJsonConverter {
         }
         case STRUCT: {
           Struct struct = (Struct) value;
-          if (struct.schema() != schema)
+          if (!struct.schema().equals(schema))
             throw new DataException("Mismatching schema.");
           ObjectNode obj = JsonNodeFactory.instance.objectNode();
           for (Field field : schema.fields()) {

--- a/kafka-connect-common/src/test/java/com/datamountaineer/streamreactor/connect/json/SimpleJsonConverterTest.java
+++ b/kafka-connect-common/src/test/java/com/datamountaineer/streamreactor/connect/json/SimpleJsonConverterTest.java
@@ -31,8 +31,7 @@ import java.math.BigInteger;
 import java.nio.ByteBuffer;
 import java.util.*;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
 
 public class SimpleJsonConverterTest {
 
@@ -43,7 +42,7 @@ public class SimpleJsonConverterTest {
   @Test
   public void booleanToJson() {
     JsonNode converted = converter.fromConnectData(Schema.BOOLEAN_SCHEMA, true);
-    assertEquals(true, converted.booleanValue());
+    assertTrue(converted.booleanValue());
   }
 
   @Test
@@ -147,14 +146,14 @@ public class SimpleJsonConverterTest {
 
 
   @Test
-  public void decimalToJson() throws IOException {
+  public void decimalToJson() {
     BigDecimal expectedDecimal = new BigDecimal(new BigInteger("156"), 2);
     JsonNode converted = converter.fromConnectData(Decimal.schema(2), expectedDecimal);
     assertEquals(expectedDecimal, converted.decimalValue());
   }
 
   @Test
-  public void dateToJson() throws IOException {
+  public void dateToJson() {
     GregorianCalendar calendar = new GregorianCalendar(1970, Calendar.JANUARY, 1, 0, 0, 0);
     calendar.setTimeZone(TimeZone.getTimeZone("UTC"));
     calendar.add(Calendar.DATE, 10000);
@@ -166,7 +165,7 @@ public class SimpleJsonConverterTest {
   }
 
   @Test
-  public void timeToJson() throws IOException {
+  public void timeToJson() {
     GregorianCalendar calendar = new GregorianCalendar(1970, Calendar.JANUARY, 1, 0, 0, 0);
     calendar.setTimeZone(TimeZone.getTimeZone("UTC"));
     calendar.add(Calendar.MILLISECOND, 14400000);
@@ -178,7 +177,7 @@ public class SimpleJsonConverterTest {
   }
 
   @Test
-  public void timestampToJson() throws IOException {
+  public void timestampToJson() {
     GregorianCalendar calendar = new GregorianCalendar(1970, Calendar.JANUARY, 1, 0, 0, 0);
     calendar.setTimeZone(TimeZone.getTimeZone("UTC"));
     calendar.add(Calendar.MILLISECOND, 2000000000);
@@ -195,7 +194,7 @@ public class SimpleJsonConverterTest {
   public void nullSchemaAndPrimitiveToJson() {
     // This still needs to do conversion of data, null schema means "anything goes"
     JsonNode converted = converter.fromConnectData(null, true);
-    assertEquals(true, converted.booleanValue());
+    assertTrue(converted.booleanValue());
   }
 
   @Test
@@ -253,6 +252,21 @@ public class SimpleJsonConverterTest {
   public void noSchemaToJson() {
     JsonNode converted = converter.fromConnectData(null, true);
     assertTrue(converted.isBoolean());
-    assertEquals(true, converted.booleanValue());
+    assertTrue(converted.booleanValue());
+  }
+
+  @Test
+  public void identicalSchemasShouldNotMismatch() {
+    Schema schema1 = SchemaBuilder.struct().field("myString", Schema.STRING_SCHEMA).build();
+    Schema schema2 = SchemaBuilder.struct().field("myString", Schema.STRING_SCHEMA).build();
+
+    assertEquals(schema1, schema2);
+
+    Struct struct1 = new Struct(schema1);
+    struct1.put("myString", "testString");
+
+    JsonNode converted = converter.fromConnectData(schema2, struct1);
+
+    assertEquals("{\"myString\":\"testString\"}", converted.toString());
   }
 }

--- a/project/Dependencies.scala
+++ b/project/Dependencies.scala
@@ -493,6 +493,7 @@ trait Dependencies {
     jerseyCommon,
     avro4s,
     kafkaClients(kafkaVersion),
+    junit,
   ) ++ enumeratum ++ circe ++ http4s
 
   //Specific modules dependencies

